pkg/codegen/nodejs/gen.go

```diff
@@ -1537,77 +1537,141 @@ func (mod *modContext) genConfig(w io.Writer, variables []*schema.Property) erro
 	return nil
 }
 
-func (mod *modContext) getRelativePath() string {
-	rel, err := filepath.Rel(mod.mod, "")
+// getRelativePath returns a path to the top level of the package
+// relative to directory passed in. You must pass in the name
+// of a directory. If you provide a file name, like "index.ts", it's assumed
+// to be a directory named "index.ts".
+// It's a thin wrapper around the standard library's implementation.
+func getRelativePath(dirname string) string {
+	var rel, err = filepath.Rel(dirname, "")
 	contract.Assert(err == nil)
 	return path.Dir(filepath.ToSlash(rel))
 }
 
-func (mod *modContext) sdkImports(nested, utilities bool) []string {
-	imports := []string{"import * as pulumi from \"@pulumi/pulumi\";"}
+// The parameter dirRoot is used as the relative path
+func (mod *modContext) getRelativePath() string {
+	return getRelativePath(mod.mod)
+}
 
-	relRoot := mod.getRelativePath()
+// sdkImports generates the imports at the top of a source file.
+// This function is only intended to be called from resource files and
+// at the index. For files nested in the `/types` folder, call
+// sdkImportsForTypes instead.
+func (mod *modContext) sdkImports(nested, utilities bool) []string {
+	return mod.sdkImportsWithPath(nested, utilities, mod.mod)
+}
+
+// sdkImportsWithPath generates the import functions at the top of each file.
+// If nested is true, then the file is assumed not to be at the top-level i.e.
+// it's located in a subfolder of the root, perhaps deeply nested.
+// If utilities is true, then utility functions are imported.
+// dirpath injects the directory name of the input file relative to the root of the package.
+func (mod *modContext) sdkImportsWithPath(nested, utilities bool, dirpath string) []string {
+	// All files need to import the SDK.
+	var imports = []string{"import * as pulumi from \"@pulumi/pulumi\";"}
+	var relRoot = getRelativePath(dirpath)
+	// Add nested imports if enabled.
 	if nested {
-		imports = append(imports, []string{
-			fmt.Sprintf(`import * as inputs from "%s/types/input";`, relRoot),
-			fmt.Sprintf(`import * as outputs from "%s/types/output";`, relRoot),
-		}...)
-
-		if mod.pkg.Language["nodejs"].(NodePackageInfo).ContainsEnums {
-			code := `import * as enums from "%s/types/enums";`
-			if lookupNodePackageInfo(mod.pkg).UseTypeOnlyReferences {
-				code = `import type * as enums from "%s/types/enums";`
-			}
-			imports = append(imports, fmt.Sprintf(code, relRoot))
-		}
+		imports = append(imports, mod.genNestedImports(relRoot)...)
 	}
-
+	// Add utility imports if enabled.
 	if utilities {
-		imports = append(imports, mod.utilitiesImport())
+		imports = append(imports, mod.utilitiesImport(relRoot))
 	}
 
 	return imports
 }
 
-func (mod *modContext) utilitiesImport() string {
-	relRoot := mod.getRelativePath()
+func (mod *modContext) utilitiesImport(relRoot string) string {
 	return fmt.Sprintf("import * as utilities from \"%s/utilities\";", relRoot)
 }
```

Comment:
Is there a benefit to keeping this a method?  Or do we just wish to preserve the association between modContext and this function?

Reply:
Not really. No reason it can't be a function.